### PR TITLE
Updating to go 1.18

### PR DESCRIPTION
**Pull request by jmurret**
Changes proposed in this PR:
- update all go versions 1.18
- change golangci-lint version to 1.46.2

How I've tested this PR:
- 👀 
- CI

How I expect reviewers to test this PR:
👀 

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 ## UNRELEASED
+BREAKING CHANGES:
+* Control Plane
+  * Update minimum go version for project to 1.18 [[GH-1292](https://github.com/hashicorp/consul-k8s/pull/1292)]
+* CLI
+  * Update minimum go version for project to 1.18 [[GH-1292](https://github.com/hashicorp/consul-k8s/pull/1292)]
+
```
Comment (ishustava) [inline CHANGELOG.md:7]: why is it a breaking change?

File: charts/consul/test/docker/Test.dockerfile [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # a script to configure kubectl, potentially install Helm, and run the tests
 # manually. This image only has the dependencies pre-installed.
 
-FROM circleci/golang:1.17
+FROM circleci/golang:1.18
```
Comment (ishustava) [inline charts/consul/test/docker/Test.dockerfile:9]: We need to build and upload a new docker image

File: hack/aws-acceptance-test-cleanup/go.mod [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
-go 1.16
+go 1.18
 
 require (
 	github.com/aws/aws-sdk-go v1.38.63
 	github.com/cenkalti/backoff/v4 v4.1.1
+)
+
+require (
```
Comment (ishustava) [inline hack/aws-acceptance-test-cleanup/go.mod:10]: why do we need two require blocks?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ env:
   TEST_RESULTS: /tmp/test-results # path to where test results are saved
   CONSUL_VERSION: 1.13.0-alpha1 # Consul's OSS version to use in tests
   CONSUL_ENT_VERSION: 1.13.0-alpha1+ent # Consul's enterprise version to use in tests
-  GOTESTSUM_VERSION: 1.6.4 # You cannot use environment variables with workflows. The gotestsum version is hardcoded in the reusable workflows too.
+  GOTESTSUM_VERSION: 1.8.1 # You cannot use environment variables with workflows. The gotestsum version is hardcoded in the reusable workflows too.
 
 jobs:
   get-go-version:
```
Comment (ishustava) [inline .github/workflows/test.yml:12]: I think we also need to update `.goversion` file as that's what's used by the `test` GHA pipeline

File: CHANGELOG.md (hunk 1 above)
Comment (ishustava) [reply CHANGELOG.md:7]: yeah I think so, if there are no known breaking changes to the user

File: hack/aws-acceptance-test-cleanup/go.mod (hunk 3 above)
Comment (jmurret) [reply hack/aws-acceptance-test-cleanup/go.mod:10]: https://github.com/hashicorp/consul-k8s/pull/878#discussion_r758834166

File: charts/consul/test/docker/Test.dockerfile (hunk 2 above)
Comment (jmurret) [reply charts/consul/test/docker/Test.dockerfile:9]: Are there instructions for this somewhere?

File: .github/workflows/test.yml (hunk 4 above)
Comment (jmurret) [reply .github/workflows/test.yml:12]: 👍 `.go-version` is updated.

File: charts/consul/test/docker/Test.dockerfile (hunk 2 above)
Comment (ndhanushkodi) [reply charts/consul/test/docker/Test.dockerfile:9]: I'm not sure where the instructions are, you could try asking ashwin, he did the last bump of our helm test dockerfile: https://github.com/hashicorp/consul-k8s/pull/1177/files

**Review by ndhanushkodi — APPROVED** [review]

**Pull request merged**